src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1844,6 +1934,182 @@ export class StudyViewPageStore {
 
         formSubmit('index.do', formOps, "_blank", 'post');
     }
+
+    @action
+    setCustomChartFilters(chartMeta: ChartMeta, values: string[]) {
```

Comment:
this method is too long.  impossible to read/understand/test.   i'm pretty sure this should be a switch(chartMeta.uniqueKey).  then call nicely named methods to reduce sample collections